flowmachine/flowmachine/utils.py

```diff
@@ -474,15 +474,42 @@ def calculate_dependency_graph(query_obj, analyse=False):
 
     Examples
     --------
+
+    A useful way to visualise the dependency graph in a Jupyter notebook is to
+    export it to an SVG string and display it directly in the notebook:
+
+    >>> import flowmachine
+    >>> from flowmachine.features import daily_location
+    >>> from flowmachine.utils import calculate_dependency_graph
+    >>> from io import BytesIO
+    >>> flowmachine.connect(flowdb_user="flowdb", flowdb_password="flowflow", redis_password="fm_redis")
+    >>> dl = daily_location(date="2016-01-01")
+    >>> G = calculate_dependency_graph(dl, analyse=True)
+    >>> A = nx.nx_agraph.to_agraph(G)
+    >>> svg_str = BytesIO()
+    >>> A.draw(svg_str, format="svg", prog="dot")
+    >>> svg_str = svg_str.getvalue().decode("utf8")
+    >>> SVG(svg_str)  # within a Jupyter notebook this will be displayed as a graph
+
+    Alternatively, you can export the dependency graph to a .dot file as follows:
+
     >>> import flowmachine
-    >>> flowmachine.connect()
     >>> from flowmachine.features import daily_location
-    >>> g = daily_location("2016-01-01").dependency_graph()
-    >>> from networkx.drawing.nx_agraph import write_dot
-    >>> write_dot(g, "daily_location_dependencies.dot")
-    >>> g = daily_location("2016-01-01").dependency_graph(True)
     >>> from networkx.drawing.nx_agraph import write_dot
-    >>> write_dot(g, "daily_location_dependencies_runtimes.dot")
+    >>> flowmachine.connect()
+    >>> G = daily_location("2016-01-01").dependency_graph()
+    >>> write_dot(G, "daily_location_dependencies.dot")
+    >>> G = daily_location("2016-01-01").dependency_graph(True)
+    >>> write_dot(G, "daily_location_dependencies_runtimes.dot")
+
+    The resulting .dot file then be converted to a .pdf file using the external
+    tool `dot` which comes as part of the [GraphViz] package:
```

Comment:
```suggestion
    tool `dot` which comes as part of the [GraphViz](https://www.graphviz.org/) package:<br />
```

Will do it, I think. Looks like footnotes don't get parsed in the Examples section, otherwise you could do `Graphviz [1]_` and then have the link in references as `.. [1]: https://graphiviz.org`

Reply:
Awesome, thanks! 👍